### PR TITLE
Fixes Deprecation warning for AssertEquals in tests

### PR DESCRIPTION
**Pull request by sebgoa**
The CloudStack fixes are left to a patch I submitted on jira for libcloud-391

Note that the github trunk does not pass tox, but the apache trunk does...

### CODE REVIEW

File: libcloud/test/compute/test_ec2.py [hunk 1]
```diff
@@ -337,7 +337,9 @@ def test_ex_describe_tags(self):
 
     def test_ex_import_keypair_from_string(self):
         path = os.path.join(os.path.dirname(__file__), "fixtures", "misc", "dummy_rsa.pub")
-        key = self.driver.ex_import_keypair_from_string('keypair', open(path).read())
+        fh = open(path)
```
Comment (Kami) [inline libcloud/test/compute/test_ec2.py]: Please use `with` context manager here.

Don't forgot to add `from __future__ import with_statement` to the to.

**Comment by Kami** [comment]
Besides the with thing, patch LGTM.

**Comment by sebgoa** [comment]
Ok here is the fix for the with statement and correction of an error in the earlier commit. This should merge fine.

**Comment by Kami** [comment]
Thanks.

For some reason second part didn't want to apply cleanly so I just manually made changes in the second commit.

``` bash
Applying: Fixes Deprecation warning for AssertEquals in tests
Applying: with statement and fix in storage/cloudfiles
error: patch failed: libcloud/test/compute/test_ec2.py:12
error: libcloud/test/compute/test_ec2.py: patch does not apply
Patch failed at 0002 with statement and fix in storage/cloudfiles
The copy of the patch that failed is found in:
   /w/lc/libcloud/.git/rebase-apply/patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
```

**Comment by Kami** [comment]
Btw, this patch has been merged so you can close the pr.

**Comment by sebgoa** [comment]
thanks

**Pull request closed without merging**